### DISCUSSION TITLE
Feature request: Trim Snap

### DISCUSSION THREAD

**Discussion by RSR1337** [rodit/SnapMod#165] [Ideas & Feature Suggestions]
Is it possible to add Disable Snap Splitting and Bypass video length on Trim Snap ? 
![Screenshot_20220526-204520_Snapchat](https://user-images.githubusercontent.com/52901789/170538143-2fd25f9e-94a2-4840-81a8-24baba3018a3.png)

can you take a look @rodit

**Comment by rodit** [comment]
This already works

**Reply by RSR1337** [reply]
> This already works

Disable Snap Splitting and Bypass video length enabled 
, but still can't upload more than 1min video, any idea?
![Screenshot_20220528-034931_Snapchat~2](https://user-images.githubusercontent.com/52901789/170801200-74e0e08c-7332-4b1d-94aa-6564bbc15fe7.png)

**Comment by Magna-Graecia** [comment]
Works for me as well. Just tested with a video that was 1 minute and 5 seconds long.

**Reply by RSR1337** [reply]
> Works for me as well. Just tested with a video that was 1 minute and 5 seconds long.

What's you settings?

**Reply by Magna-Graecia** [reply]
Disable Snap Splitting and Bypass Video Length Restrictions both enabled. I shared the video from gallery to Snapchat (easiest way to get it loaded into the camera edit window).

**Reply by RSR1337** [reply]
> Disable Snap Splitting and Bypass Video Length Restrictions both enabled. I shared the video from gallery to Snapchat (easiest way to get it loaded into the camera edit window).

I heard from my friends, when you upload a video from trim snap, the video quality doesn't decrease, I'm not sure if it's true or not

**Reply by Magna-Graecia** [reply]
Trim snap is an Xposed module or apk or something? Or you mean after trimming a snap in Snapchat the video will not be compressed?

**Reply by RSR1337** [reply]
> Trim snap is an Xposed module or apk or something? Or you mean after trimming a snap in Snapchat the video will not be compressed?

This 
![Snapchat-371327107](https://user-images.githubusercontent.com/52901789/170802539-1bf180df-cf5f-4817-ae26-ba774524f649.jpg)

**Reply by Magna-Graecia** [reply]
That's Timeline Mode, not trim snap. You can trim a snap using the rectangular box that shows in the bottom left of the camera edit window. I tried timeline mode, but I don't see a point to it, especially when sending as a live snap. It's basically the same as Multi-Snap, but for videos.

**Comment by RSR1337** [comment]
I thought the long video would be split into 60 seconds ، but after testing it works, thanks for the reply 🙏

**Reply by Magna-Graecia** [reply]
Nope. Disable Snap Splitting just allows you to send any length video without it being chopped into separate 10-second snaps, and therefore no video or audio interruptions when viewing. Bypass Video Length restrictions allows you to load any length video into the camera edit window, and also to send them, and you will be able to trim it as normal. Glad it worked out for you :smile:
